src/jarabe/frame/clipboardmenu.py

```diff
@@ -176,13 +176,19 @@ def _journal_item_activate_cb(self, menu_item):
         jobject = self._copy_to_journal()
         jobject.destroy()
 
-    def _write_to_temp_file(self, data):
+    def _write_to_temp_file(self, data, is_pixbuf):
         tmp_dir = os.path.join(env.get_profile_path(), 'data')
         f, file_path = tempfile.mkstemp(dir=tmp_dir)
+
         try:
-            os.write(f, data)
+            if not is_pixbuf:
+                os.write(f, data)
+            else:
+                _type = self._cb_object.get_mime_type().split('/').pop()
+                data.savev(file_path, _type, [], [])
         finally:
-            os.close(f)
+            if not is_pixbuf:
+                os.close(f)
```

Comment:
I am not really sure, shouldn't we close the file even if we are not writing to; since `tempfile.mkstemp` retrurns an open file handle? Or is it fine? Also, if we have to close it, should we close the file before the `data.savev` function is called?

Reply:
Agreed, I'd gotten an error earlier as I didn't see a reason not to close the file too and I can't remember the error and it seems it no longer happens, seeing as the `close` call is in a finally block, all operations happen before it.